src/main/java/io/cdap/plugin/google/drive/common/GoogleDriveClient.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.drive.common;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import io.cdap.plugin.google.drive.common.exceptions.InvalidPropertyTypeException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * Base client for working with Google Drive API
+ *
+ * @param <C> configuration
+ */
+public abstract class GoogleDriveClient<C extends GoogleDriveBaseConfig> {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String ROOT_FOLDER_ID = "root";
+  protected static final String FULL_PERMISSIONS_SCOPE = "https://www.googleapis.com/auth/drive";
+  protected Drive service;
+  protected final C config;
+  private NetHttpTransport httpTransport;
+
+  public GoogleDriveClient(C config) {
+    this.config = config;
+    try {
+      httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    } catch (Exception e) {
+      throw new RuntimeException("There was issue during communicating with Google Drive API.", e);
+    }
+    service = new Drive.Builder(httpTransport, JSON_FACTORY, getCredentials())
+      .build();
+  }
+
+  private Credential getCredentials() {
```

Comment:
add a javadoc with exceptions being thrown from here and handle them appropriately on caller side.

Reply:
Done.